components/reference-data/src/main/java/org/datacleaner/beans/transform/RemoveDictionaryMatchesTransformer.java

```diff
@@ -98,27 +135,64 @@ public void close() {
     @Override
     public Object[] transform(InputRow inputRow) {
         final String value = inputRow.getValue(_column);
-        final String result = transform(value);
-        return new Object[] { result };
+        final Object[] result = transform(value);
+        return result;
     }
 
-    public String transform(final String value) {
+    public Object[] transform(final String value) {
+
         if (Strings.isNullOrEmpty(value)) {
-            return value;
+            switch (_removedMatchesType) {
+            case STRING:
+                return new String[] { value, "" };
+            case LIST:
+                return new Object[] { value, Collections.EMPTY_LIST };
+            }
+        }
+        final Object removedStringMatches;
+        switch (_removedMatchesType) {
+        case STRING:
+            removedStringMatches = new StringBuilder();
+            break;
+        case LIST:
+            removedStringMatches = new ArrayList<String>();
+            break;
+        default:
+            removedStringMatches = null;
         }
 
-        final StringBuilder sb = new StringBuilder();
+        final StringBuilder survivorString = new StringBuilder();
         final Iterable<String> tokens = SPLITTER.split(value);
         for (String token : tokens) {
             if (!dictionaryConnection.containsValue(token)) {
-                if (sb.length() != 0) {
-                    sb.append(' ');
+                if (survivorString.length() != 0) {
+                    survivorString.append(' ');
                 }
-                sb.append(token);
+                survivorString.append(token);
+            } else {
+                addRemovedMatches(removedStringMatches, token);
             }
         }
 
-        return sb.toString();
+        switch (_removedMatchesType) {
+        case STRING:
+            return new String[] { survivorString.toString(), removedStringMatches.toString() };
+        case LIST:
+            return new Object[] { survivorString.toString(), removedStringMatches };
+        }
+        return null;
     }
 
+    @SuppressWarnings("unchecked")
+    public void addRemovedMatches(Object removedObjects, final String token) {
+        if (removedObjects instanceof StringBuilder) {
+            StringBuilder removedString = (StringBuilder) removedObjects;
+            if (removedString.length() != 0) {
+                removedString.append(' ');
+            }
+            removedString.append(token);
+        } else if (removedObjects instanceof ArrayList) {
+            ((ArrayList<String>) removedObjects).add(token);
+        }
+    }
```

Comment:
This is... Honestly, pretty nasty. I'd consider using overloading, if using the Object class for removedObjectMatches is absolutely needed (which I would try very, very hard to avoid).